pkg/handler/enqueue_owner.go

```diff
@@ -112,6 +115,7 @@ func (e *enqueueRequestForOwner) Delete(ctx context.Context, evt event.DeleteEve
 
 // Generic implements EventHandler.
 func (e *enqueueRequestForOwner) Generic(ctx context.Context, evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	e.logEvent("Generic", evt.Object, nil)
```

Comment:
It adds some amount of processing to a code path, which is frequently called. I am wondering whether it would not be possible to leverage what is already returned by `getOwnerReconcileRequest` a couple of lines below to get information on the owner instead of what is made in `logEvent` and `extractTypedOwnerReference`?
Also wouldn't we want to provide information on the object itself?